### ISSUE TITLE
Fix TensorFlow Lite issue on Android phone 

### ISSUE THREAD

**Issue by bearprada** [plant-tw/PlantsCam-Android#2]
# Input model 
[plant_ab.pb.zip](https://github.com/plant-tw/PlantsCam-Android/files/1821597/plant_ab.pb.zip)

# Command 

bazel-bin/tensorflow/contrib/lite/toco/toco --input_file=plant_ab.pb --output_file=plant.tflite --input_format=TENSORFLOW_GRAPHDEF --output_format=TFLITE inference_type=FLOAT --input_shapes=1,128,128,3:1:1 --input_arrays=agent/img,agent/len_pixel,agent/len_cm --output_arrays=agent/Softmax

# Error

Here is a list of operators for which you will need custom implementations: TensorFlowMax.

# Possible solution

1. Implement this operator in Custom Operator https://github.com/tensorflow/tensorflow/blob/master/tensorflow/contrib/lite/g3doc/custom_operators.md
2. Implement this operator into the build tool, ref : https://github.com/tensorflow/tensorflow/pull/17074
3. waiting the new version of Tensorflow Lite
4. replace the implementation, e.g, Caffe.

# Environment  :
* TensorFlow Lite v1.5

**Comment by bearprada** [comment]
I will go solution 1&2 first

**Comment by tcw165** [comment]
@bearprada: You could email this guy, `顧仁民 renminml@google.com` for further suggestion. He initially suggests you to go 1st solution because 2nd solution introduces the complexity. e.g. Recompile the `toco`.

**Comment by bearprada** [comment]
thx for your help :) @boyw165 i will send email to him

**Comment by bearprada** [comment]
based on this video https://www.youtube.com/watch?v=kFwVOxBOlsU
the custom operators should be implemented in this way

<img width="528" alt="screen shot 2018-04-07 at 10 38 15 am" src="https://user-images.githubusercontent.com/1080103/38450411-d0f830b0-3a4f-11e8-86ea-2c82b00a87e1.png">

Which is using the C++ API on Android App.

**Comment by bearprada** [comment]
In the end, I decide to use `Tensorflow Mobile` (it's not Tensfolow Lite) for the first version of implementation on Android.

because they provide more operator, so I can use FreezeGraph directly, don't need to spend time on the custom operator for Tensorflow Lite :) which means I don't need to write any C++ code this time.

**Comment by bearprada** [comment]
commit : d358c1ba0d0d1b4eac21617b9a7acfc68fae8c60

**Comment by ghost** [comment]
@bearprada, feel free to send me mails. We are supporting more ops now and external feedback is valuable. I remember reduce_max is supported.

Are you working on some project to classify plant? Interesting project.

**Comment by bearprada** [comment]
hi @renminio sorry for the reply later. 

yes, we are working on a plant detection project with an organization called g0v in Taiwan ( https://g0v.tw/en-US/ ). here is the proposal (it's Chinese ) 

https://grants.g0v.tw/projects/5a49255d8b4d8d001efe2e6a

and this is current status(Chinese) https://drive.google.com/file/d/1-ZHZlsYQc6ema4zfAk906Gr1rDCXIiQr/view

We collect 79 species in Taipei Daan Park, the accuracy is `97%` for training data, `83%` for validation data.

**Comment by bearprada** [comment]
BTW, we retrained the model based on MobileNet, and it can easy to convert to TFlite.
